modules/setting/setting.go

```diff
@@ -801,21 +804,8 @@ func NewContext() {
 		return
 	}
 
-	if OAuth2.Enable {
-		OAuth2.JWTSecretBytes = make([]byte, 32)
-		n, err := base64.RawURLEncoding.Decode(OAuth2.JWTSecretBytes, []byte(OAuth2.JWTSecretBase64))
-
-		if err != nil || n != 32 {
-			OAuth2.JWTSecretBase64, err = generate.NewJwtSecret()
-			if err != nil {
-				log.Fatal("error generating JWT secret: %v", err)
-				return
-			}
-
-			CreateOrAppendToCustomConf(func(cfg *ini.File) {
```

Comment:
should we realy drop this?

Reply:
It's not dropped. You can find it in `loadOrCreateSymmetricKey`

Reply:
without this no `JWT_SECRET` will be created if there is none

Reply:
so it's created on demand ... I would prefer if it get added on first gitea start ...

Reply:
It gets added on first gitea start if you use `HSxxx` as signing algorithm. If you use another algorithm it is not used at all.